Algorithms/Numeric/Factorial.cs

```diff
@@ -11,16 +12,18 @@ public static class Factorial
         /// <summary>
         ///     Calculates factorial of a number.
         /// </summary>
-        /// <param name="num">Input number.</param>
+        /// <param name="num">Input number. Caller responsible for num >= 0.</param>
         /// <returns>Factorial of input number.</returns>
-        public static long Calculate(int num)
+        public static BigInteger Calculate(BigInteger num)
```

Comment:
Could you make it generic to work with INumber constraint? That way you wouldn't need to copy it in Maclaurin